pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,333 @@
+	Labels Labels
+	// Annotations is an unstructured key value map that may be set by external
+	// tools to store and retrieve arbitrary metadata.
+	Annotations map[string]string
+	// Privileged runs the container in the privileged mode.
+	Privileged bool
+	// ReadOnlyRootFS sets the root filesystem of the container to be
+	// read-only.
+	ReadOnlyRootFS bool
+	// Path to store the container log on the host (i.e., outside of the
+	// sandbox).
+	LogPath string
+
+	// Variables for interactive containers, these have very specialized
+	// use-cases (e.g. debugging).
+	// TODO: Determine if we need to continue supporting these fields that are
+	// part of Kubernetes's Container Spec.
+	Stdin     bool
+	StdinOnce bool
+	TTY       bool
+
+	// Linux contains configuration specific to Linux containers.
+	Linux *LinuxContainerConfig
+}
+
+// LinuxContainerConfig contains platform-specific configuration for
+// Linux-based containers.
+type LinuxContainerConfig struct {
+	// Resources specification for the container.
+	Resources LinuxContainerResources
+	// AddCapabilities lists capabilities to add.
+	AddCapabilities []string
+	// DropCapabilities lists capabilities to drop.
+	DropCapabilities []string
+	// SELinux is the SELinux context to be applied.
+	SELinux *api.SELinuxOptions
+}
+
+// LinuxContainerResources specifies Linux specific configuration for
+// resources.
+// TODO: Consider using Resources from opencontainers/runtime-spec/specs-go
+// directly.
+type LinuxContainerResources struct {
+	// CPU CFS (Completely Fair Scheduler) period
+	CPUPeriod uint64
+	// CPU CFS (Completely Fair Scheduler) quota
+	CPUQuota uint64
+	// CPU shares (relative weight vs. other containers)
+	CPUShares uint64
+	// Memory limit in bytes
+	MemoryLimitInBytes uint64
+	// OOMScoreAdj adjusts the oom-killer score.
+	OomScoreAdj int
+	// Swappiness specifies how aggressive the kernel will swap memory pages.
+	// Range from 0 to 100.
+	Swappiness uint64
+}
+
+// ContainerFilter is used to filter containers.
+type ContainerFilter struct {
+	// Name of the container.
+	Name string
+	// ID of the container.
+	ID RawContainerID
+	// State of the contianer.
+	State ContainerState
+	// LabelSelector to select matches.
+	// Only api.MatchLabels is supported for now and the requirements
+	// are ANDed. MatchExpressions is not supported yet.
+	LabelSelector unversioned.LabelSelector
+}
+
+type StreamOptions struct {
+	TTY          bool
+	InputStream  io.Reader
+	OutputStream io.Writer
+	ErrorStream  io.Writer
+}
+
+// KeyValue represents a key-value pair.
+type KeyValue struct {
+	Key   string
+	Value string
+}
+
+// ImageOperations offers basic image operations.
+type ImageOperations interface {
+	// List lists the existing images.
+	List() ([]Image, error)
+	// Pull pulls an image with authentication config.
+	Pull(image ImageSpec, auth AuthConfig) error
+	// Remove removes an image.
+	Remove(image ImageSpec) error
```

Comment:
Image apis take an ImageSpec (instead of image id) because it is more easier to take in new image formats.